Recrutify/Recrutify.Services/Services/ScheduleService.cs

```diff
@@ -80,9 +92,80 @@ public async Task UpdateScheduleSlotsForCurrentUserAsync(IEnumerable<DateTime> d
             await _scheduleRepository.BulkUpdateScheduleSlotsAsync(currentUserId, newListDateTime, removedListDateTime);
         }
 
-        private IEnumerable<DateTime> GetDateTimeInScheduleSlots(IEnumerable<ScheduleSlot> scheduleSlots)
+        public async Task ProcessingAppointOrCancelInterviewsAsync(IEnumerable<InterviewDTO> interviews, Guid projectId, CancellationToken cancellationToken)
         {
-            return scheduleSlots?.Select(s => s.AvailableTime) ?? new List<DateTime>();
+            var projectExists = await _projectService.ExistsAsync(projectId, cancellationToken);
+            if (!projectExists)
+            {
+                throw new ValidationException("Project does not exist.");
+            }
+
+            var allCandidates = await _candidateService.GetCandidatesByIdsAsync(interviews.Select(i => i.CandidateId));
+            var dictionaryInterviews = _mapper.Map<IEnumerable<Interview>>(interviews)
+                                                .GroupBy(i => i.IsAppointment)
+                                                .ToDictionary(k => k.Key, v => v.Select(i => i).ToList());
```

Comment:
write just v.ToList(), select not necessary

Reply:
+